src/protocol/gossip/impl/connectivity.cpp

```diff
@@ -403,28 +445,32 @@ namespace libp2p::protocol::gossip {
         flat_changes.emplace_back(p.second, std::move(p.first));
       });
 
-      // clang-format off
       connected_peers_.selectAll(
-          [&flat_changes, this] (const PeerContextPtr& ctx) {
-            boost::for_each(flat_changes, [&ctx] (auto&& p) {
-                ctx->message_builder->addSubscription(p.first, p.second);
+          [&flat_changes, this](const PeerContextPtr &ctx) {
+            boost::for_each(flat_changes, [&ctx](auto &&p) {
+              ctx->message_builder->addSubscription(p.first, p.second);
             });
             flush(ctx);
-          }
-       );
+          });
 
     } else {
       flush();
       writable_peers_on_heartbeat_.selectAll(
-          [this](const PeerContextPtr& ctx) {
-            flush(ctx);
-          }
-      );
+          [this](const PeerContextPtr &ctx) { flush(ctx); });
     }
-    // clang-format on
 
     writable_peers_low_latency_.clear();
     writable_peers_on_heartbeat_.clear();
+
+    if (ts >= addresses_renewal_time_) {
+      addresses_renewal_time_ = scheduler_->now()
+          + config_.address_expiration_msec - std::chrono::milliseconds(500);
```

Comment:
Not clear why we are subtracting 500ms here. Probably this should be a named constant with a comment.